### PR TITLE
RSDK-9046 Clearer phone proxy code

### PR DESCRIPTION
**Pull request by benjirewis**

### CODE REVIEW

File: phone_proxy/analysis_options.yaml [hunk 1]
```diff
@@ -1,28 +1,4 @@
-# This file configures the analyzer, which statically analyzes Dart code to
```
Comment (benjirewis) [inline phone_proxy/analysis_options.yaml:1]: Removed a lot of autogenerated comments in config files.

File: phone_proxy/ios/Runner/AppDelegate.swift [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import Flutter
 import UIKit
 
-@UIApplicationMain
+@main
```
Comment (benjirewis) [inline phone_proxy/ios/Runner/AppDelegate.swift:4]: `flutter run` always makes this change on my behalf 🤷🏻 .

File: phone_proxy/lib/main.dart
Comment (benjirewis) [inline phone_proxy/lib/main.dart:1]: This file is going to be hard to review with a combined diff. Use a split diff or look at the new file directly. It's extremely similar code, but I've moved pieces around quite a bit. I'll also leave comments pointing out non-obvious changes I've thrown in.

File: phone_proxy/lib/main.dart [hunk 3]
```diff
@@ -1,149 +1,231 @@
...
-import 'package:ble/ble_central.dart';
-import 'package:ble/ble_peripheral.dart';
-import 'package:ble/ble_socket.dart';
+import 'package:blev/ble.dart';
+import 'package:blev/ble_central.dart';
+import 'package:blev/ble_peripheral.dart';
+import 'package:blev/ble_socket.dart';
+
 import 'package:flutter/material.dart';
+import 'package:logger/logger.dart';
```
Comment (benjirewis) [inline phone_proxy/lib/main.dart:11]: I switched from `print`s that show up in the actual app to an actual Flutter logger. Reasoning being that in an eventual customer app, they probably won't want printing. If they want to share the content of some logs as actual UI messages, that should be their decision.

File: phone_proxy/lib/main.dart [hunk 4]
```diff
@@ -1,149 +1,231 @@
...
+  //   device will need to write its `mobileDevice` value
+  // - Ready to establish an L2CAP connection for SOCKS forwarding once a value
+  //   is written to the above characteristic
+  //
+  // You may want to ask users to enter the `machineToManage` value as part
+  // of app setup.
+  var machineToManage = 'mac1.loc1.viam.cloud';
 
-var machineToManage = 'mac1.loc1.viam.cloud';
+  runZonedGuarded(
```
Comment (benjirewis) [inline phone_proxy/lib/main.dart:83]: `runZoned` -> `runZonedGuarded` due to some advice I read online and a deprecation warning for `runZoned`.

File: phone_proxy/lib/main.dart [hunk 5]
```diff
@@ -1,149 +1,231 @@
...
+      runApp(const MyApp());
+    }, (error, stackTrace) {
+      logger.e('error running BLE-SOCKS phone proxy: $error');
+    },
+  );
+
+  // TODO(benji): correct this section.
+  //
+  // If you would like to stop the phone proxy and restart it with a new `machineToManage`
+  // value. Use the following code:
```
Comment (benjirewis) [inline phone_proxy/lib/main.dart:95]: Would love some advice here; customer wanted the ability to change the `machineToManage` value without restarting the app. Not quite sure how to do it.

File: phone_proxy/lib/main.dart [hunk 6]
```diff
@@ -1,149 +1,231 @@
...
-              streamSub.cancel();
-              manageMachine(bleCentral, machineToManage);
-            }
-          });
-        });
-      }).catchError((error) {
-        print('error requesting bluetooth permissions $error');
-      });
+// Give some BLE operations a few retries for resiliency.
+const numRetries = 3;
```
Comment (benjirewis) [inline phone_proxy/lib/main.dart:112]: I tried to add retries in a couple of places you left `TODO`s (particularly in discovering services and characteristics).

File: phone_proxy/lib/main.dart [hunk 7]
```diff
@@ -1,149 +1,231 @@
...
+          logger.e(
               'did not find needed periph name char after discovery; disconnecting');
           await periph.disconnect();
           deviceSub.resume();
           return;
         }
 
         final periphName = utf8.decode((await periphNameChar.read())!);
-        if (periphName != machineName) {
-          // Note(erd): this could use some retry logic
```
Comment (benjirewis) [inline phone_proxy/lib/main.dart:145]: Did _not_ add retry logic to this one (reading and decoding the char value). Once the characteristic has been discovered, is there really a case where the value might not yet be readable?

File: phone_proxy/pubspec.yaml [hunk 8]
```diff
@@ -1,95 +1,31 @@
...
   permission_handler: ^11.3.1
   socks5_proxy: ^1.0.6
-  ble:
-    # Temporary until public
-    git: git@github.com:viamrobotics/flutter-ble.git
+  # `blev` is Viam's underlying library for BLE-SOCKS interactions (called
+  # `flutter-ble` on github). We recommend that you do not interface with the
+  # library directly and instead use a copy of the API exposed in this example
+  # app.
+  blev: ^0.0.2
```
Comment (benjirewis) [inline phone_proxy/pubspec.yaml:21]: `blev` now [public](https://pub.dev/packages/blev/versions/0.0.2) 🎉 .

File: phone_proxy/test/widget_test.dart
Comment (benjirewis) [inline phone_proxy/test/widget_test.dart:1]: Unnecessary default test?

Comment (edaniels) [reply phone_proxy/test/widget_test.dart:1]: yep

File: phone_proxy/lib/main.dart (hunk 5 above)
Comment (edaniels) [reply phone_proxy/lib/main.dart:95]: I think this is what they should do (stop, start), but they'll need to adapt the app to their liking with some variable.

File: phone_proxy/lib/main.dart (hunk 3 above)
Comment (edaniels) [reply phone_proxy/lib/main.dart:11]: yep makes sense

**Review by edaniels — APPROVED** [review]
all LGTM, good changes!

File: phone_proxy/lib/main.dart (hunk 5 above)
Comment (benjirewis) [reply phone_proxy/lib/main.dart]: Removed this comment section.

**Pull request merged**